### ISSUE TITLE
Typo in zlib inflate?

### ISSUE THREAD

**Issue by hermanbergwerf** [brendan-duncan/archive#38]
I would think that this line: https://github.com/brendan-duncan/archive/blob/89a489c77f9d36e3f0f0def06abf15a7a0a06fa2/lib/src/zlib/inflate.dart#L202, should be `if (len != ~nlen)`. The error should be thrown if the second two bytes complement is NOT equal to the first two right? (https://www.w3.org/Graphics/PNG/RFC-1951#noncompressed). Or am I missing something?

**Comment by brendan-duncan** [comment]
So, while you were right, the line was a bug, it wasn't as simple a fix as you suggested. 
The short version: I pushed the fix.
The long version: In Dart, all ints are signed, and the ~nlen operation turned it into a signed int, so what would have been 65531 as an uint16, became -5 and the test failed, even though len was 65531. So I had to do some binary voodoo to turn the signed int back into a uint16 so that it could compare correctly with len.  Dart has many great qualities, but dealing with specific types of ints (int16, uint32, etc) is a pain...but not nearly as much of a pain as Javascript.
